External/SPMmodified/toolbox/cat12/cat_main_updateSPM.m

```diff
@@ -84,12 +84,14 @@
   %% Some error handling
   %    ds('l2','',vx_vol,Ysrc./WMth,Yp0>0.3,Ysrc./WMth,Yp0,80)
   Yp0  = single(P(:,:,:,3))/255/3 + single(P(:,:,:,1))/255*2/3 + single(P(:,:,:,2))/255;
-  if isfield(res,'Ylesion') && sum(res.Ylesion(:)>0)
-    res.YlesionFull = res.Ylesion; % ExploreASL fix
-    res.Ylesion = cat_vol_ctype( single(res.Ylesion) .* (Yp0>0.2) ); 
-    for k=1:size(P,4), Yl = P(:,:,:,k); Yl(res.Ylesion>0.5) = 0; P(:,:,:,k) = Yl; end  
-    Yl = P(:,:,:,3); Yl(res.Ylesion>0.5) = 255; P(:,:,:,3) = Yl; clear Yl; 
-    Yp0  = single(P(:,:,:,3))/255/3 + single(P(:,:,:,1))/255*2/3 + single(P(:,:,:,2))/255;
+  if isfield(res,'Ylesion') 
```

Comment:
I wonder if the behavior differs between Ylesion and YlesionFull, so why not just keeping one lesion parameter?

You probably don't want the changes here applied to YlesionFull, then perhaps better naming would be YlesionMasking.

Otherwise, if you replace all Ylesion by YlesionFull, this is tricky. Probably this code here is there for the automatically detected lesions (e.g., WMH) by CAT12, and you don't want to change this behavior in the rest of CAT12 (which we now do as you replaced Ylesion by YlesionFull in the rest).

Shouldn't we instead detect user-specific lesions, for user-specific lesions we skip this part, otherwise we run this part but we keep a single Ylesion parameter throughout?

Reply:
The point of this edit was to make sure YLesionFull is initialized with YLesion even if YLesion was empty.

YLesionFull simply is a concatenation of our own manual Lesions and the automatic YLesion from cat12, so we don't want to disrupt what is done with YLesion, but we want to have our joint YLesionFull. So the code changes might not be 100% optimal, but they are minimally invasive.

Reply:
We should keep the original variable naming of CAT12 throughout, except if we add our own Ylesion (e.g. yLesionCostMasking). I thought you now renamed all CAT12 Ylesion's? If you only renamed our variable, I agree.

Just checking, as we want to update SPM & CAT12 at some point.